test/e2e/ingress.go

```diff
@@ -0,0 +1,311 @@
+		IngressRuleValue: extensions.IngressRuleValue{
+			HTTP: &extensions.HTTPIngressRuleValue{
+				Paths: []extensions.HTTPIngressPath{
+					{
+						Path: fmt.Sprintf("/%v%d", pathPrefix, i),
+						Backend: extensions.IngressBackend{
+							ServiceName: fmt.Sprintf("%v%d", appPrefix, i),
+							ServicePort: util.NewIntOrStringFromInt(httpContainerPort),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// createIngress creates an Ingress with num rules. Eg:
+// start = 1 num = 2 will given you a single Ingress with 2 rules:
+// Ingress {
+//	 foo1.bar.com: /foo1
+//	 foo2.bar.com: /foo2
+// }
+func createIngress(c *client.Client, ns string, start, num int) extensions.Ingress {
+	ing := extensions.Ingress{
+		ObjectMeta: api.ObjectMeta{
+			Name:      fmt.Sprintf("%v%d", appPrefix, start),
+			Namespace: ns,
+		},
+		Spec: extensions.IngressSpec{
+			Backend: &extensions.IngressBackend{
+				ServiceName: fmt.Sprintf("%v%d", appPrefix, start),
+				ServicePort: util.NewIntOrStringFromInt(httpContainerPort),
+			},
+			Rules: []extensions.IngressRule{},
+		},
+	}
+	for i := start; i < start+num; i++ {
+		ing.Spec.Rules = append(ing.Spec.Rules, ruleByIndex(i))
+	}
+	Logf("Creating ingress %v", start)
+	_, err := c.Extensions().Ingress(ns).Create(&ing)
+	Expect(err).NotTo(HaveOccurred())
+	return ing
+}
+
+// createApp will create a single RC and Svc. The Svc will match pods of the
+// RC using the selector: 'name'=<name arg>
+func createApp(c *client.Client, ns string, i int) {
+	name := fmt.Sprintf("%v%d", appPrefix, i)
+	l := map[string]string{}
+
+	Logf("Creating svc %v", name)
+	svc := svcByName(name, httpContainerPort)
+	svc.Spec.Type = api.ServiceTypeNodePort
+	_, err := c.Services(ns).Create(svc)
+	Expect(err).NotTo(HaveOccurred())
+
+	Logf("Creating rc %v", name)
+	rc := rcByNamePort(name, 1, testImage, httpContainerPort, l)
+	rc.Spec.Template.Spec.Containers[0].Args = []string{
+		"--num=1",
+		fmt.Sprintf("--start=%d", i),
+		fmt.Sprintf("--prefix=%v", pathPrefix),
+		fmt.Sprintf("--port=%d", httpContainerPort),
+	}
+	_, err = c.ReplicationControllers(ns).Create(rc)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+// gcloudUnmarshal unmarshals json output of gcloud into given out interface.
+func gcloudUnmarshal(resource, regex string, out interface{}) {
+	output, err := exec.Command("gcloud", "compute", resource, "list",
+		fmt.Sprintf("--regex=%v", regex), "-q", "--format=json").CombinedOutput()
+	if err != nil {
+		Failf("%v", err)
+	}
+	if err := json.Unmarshal([]byte(output), out); err != nil {
+		Failf("%v", err)
+	}
+}
+
+func checkLeakedResources() error {
+	msg := ""
+	// TODO: These are the main resources but we should really check them all.
+	beList := []compute.BackendService{}
+	gcloudUnmarshal("backend-services", "k8s-be-[0-9]+", &beList)
+	if len(beList) != 0 {
+		for _, b := range beList {
+			msg += fmt.Sprintf("%v\n", b.Name)
+		}
+		return fmt.Errorf("Found backend services:\n%v", msg)
```

Comment:
This isn't a failure actually, it's just polling on GLBC to cleanup resources. I think it'll be weird if it keeps printing "Failed to cleanup resources" 2/3 times and then passes. I'll add an assert.